### PR TITLE
Some macros

### PR DESCRIPTION
**Pull request by kbknapp**

### CODE REVIEW

**Comment by kbknapp** [comment]
@scooter-dangle should be good now

File: src/deroff.rs [hunk 1]
```diff
@@ -1299,3 +1331,42 @@ fn test_is_white() {
...
+            panic!("from and to must be the same length");
+        }
+        for (f, t) in from.chars().zip(to.chars()) {
+            self.map.insert(f.to_string(), t.to_string());
+        }
+    }
+    fn translate(&self, s: &str) -> String {
+        let mut s = s.to_owned();
+        for (k, v) in self.map.iter() {
+            s = (&*s).replace(k, v);
```
Comment (scooter-dangle) [inline src/deroff.rs]: Would be good to include tests a couple of this logic to ensure it does. E.g., the following fails:
```rust
#[test]
fn test_translate_table() {
    let mut tbl = TranslateTable::new();
    tbl.maketrans("abc", "bca");
    assert_eq!(tbl.translate("abc"), "bca");
}
```
Not sure if any valid roff files ever have shared chars between the `to` and `from` strings, but wouldn't want to rely on that.

File: src/deroff.rs [hunk 2]
```diff
@@ -555,6 +700,20 @@ impl Deroffer {
...
+        match s.chars().nth(0) {
+            Some('.') | Some('\'') => !request_or_macro(s),
+            Some(c) => {
+                if self.tbl {
+                    do_tbl(s)
+                } else {
+                    text(s)
+                }
+            }
+            None => unreachable!(),
```
Comment (scooter-dangle) [inline src/deroff.rs]: Nit: Not locally unreachable. Maybe
```suggestion
            None => panic!("do_line` called with empty string as argument"),
```
instead?

File: src/deroff.rs (hunk 1 above)
Comment (scooter-dangle) [reply src/deroff.rs]: Obsoleted by #64

**Pull request merged**